pandas/core/groupby.py

```diff
@@ -2201,36 +2201,45 @@ def __init__(self, index, grouper=None, obj=None, name=None, level=None,
                     raise AssertionError('Level %s not in index' % str(level))
                 level = index.names.index(level)
 
-            inds = index.labels[level]
-            level_index = index.levels[level]
-
             if self.name is None:
                 self.name = index.names[level]
 
-            # XXX complete hack
+            if isinstance(index, MultiIndex):
+                inds = index.labels[level]
```

Comment:
Do you mean everything inside the `if isinstance(index, MultiIndex):` block?  If so then it looks like the MultiIndex override would need to input `grouper`, `index`, and `level` and return a tuple of `labels`, `level_index`, and `grouper`.  This seems a little messy to me since the parent Index method would have no use for the `level` parameter and would need to return None for the `labels` and `level_index` values.